src/handler.rs

```diff
@@ -400,3 +424,27 @@ fn handle_evm_withdrew<'a, C: evm::EvmChain>(
     };
     s.boxed()
 }
+
+pub fn calculate_fee(fee_percent: f64, principle: U256) -> U256 {
+    let mill_fee = (fee_percent * 1_000_000.0) as u32;
+    let mill_u256: U256 = principle * (mill_fee);
+    let fee_u256: U256 = mill_u256 / (1_000_000);
+    return fee_u256
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+
+    #[test]
+    fn percent_fee() {
+        let submitted_value: U256 = U256::from_dec_str("5000000000000000").ok().unwrap();
+        let expected_fee: U256 = U256::from_dec_str("250000000000000").ok().unwrap();
+        let withdraw_fee_percent_dec: f64 = 0.05;
+        let formatted_fee: U256 = calculate_fee(withdraw_fee_percent_dec, submitted_value);
```

Comment:
omit the types when possible.

```suggestion
        let submitted_value = U256::from_dec_str("5000000000000000").ok().unwrap();
        let expected_fee = U256::from_dec_str("250000000000000").ok().unwrap();
        let withdraw_fee_percent_dec = 0.05f64;
        let formatted_fee = calculate_fee(withdraw_fee_percent_dec, submitted_value);
```